core/src/main/scala/com/gu/tableversions/core/InMemoryTableVersions.scala

```diff
@@ -17,95 +13,63 @@ import com.gu.tableversions.core.util.RichRef._
 class InMemoryTableVersions[F[_]] private (allUpdates: Ref[F, TableUpdates])(implicit F: Sync[F])
```

Comment:
General question: is it an anticipated concern that within the same jvm process 2 or more concurrent operations might modify `TableUpdates` ? Or more a general design direction ?

Reply:
I don't know. Hence erring on the side of caution...

Reply:
Also note that this impl will only be used in tests...

Reply:
Could this be a case of defensive coding ? No need to change change as it's for test only. Just that _erring onthe side of caution_ makes it sound a bit like that.

Reply:
Perhaps, that is a fair point. But I think it's reasonable to implement everything that involves mutable state in such a way that it works correctly however it's used. On balance, I'd rather not spend the time trying to reason about whether it's safe or not, making assumptions about how it will be used in the future.